src/cpp/r/RSexp.cpp

```diff
@@ -118,11 +118,29 @@ FunctionSymbolUsageCache& functionSymbolUsageCache()
    return instance;
 }
 
+std::string translate(SEXP charSEXP, bool asUtf8 = false)
+{
+   if (asUtf8)
+   {
+      if (Rf_getCharCE(charSEXP) == CE_UTF8)
```

Comment:
This ultimately just saves us a hunt for the C string's null byte, as `Rf_translateCharUTF8()` just returns a pointer to the existing data when the string is already UTF-8 (or ASCII). But it doesn't hurt.